guides/hack/03-async/01-intro.md

```diff
@@ -1,38 +1,32 @@
 # Async
 
-Hack and PHP do not natively support multithreaded programming. However, one can envision many scenarios where such a capability might be useful. Common examples being executing two or more heavy computations; or making two or more web or database queries. 
-
-One could easily expect that the following two calls are unrelated and could be run independent of each other. 
-
-@@ intro-examples/non-async-curl.php @@
-
-In the example above, the call to `curl_exec` in `curl_A()` is blocking any other processing. Thus, even though `curl_B()` is an independent call from `curl_A()`, it has to sit around waiting for `curl_A()` to finish before beginning its execution.
-
-![No Async](/images/no-async.png)
-
 Hack provides a feature called **async** that provides your program the benefit of cooperative multi-tasking. It allows code that utilizes the async infrastructure to hide input/output (I/O) latency and data fetching. So, if you have code that has operations that involve some sort of waiting (e.g., network access, waiting for database queries), async minimizes the downtime your program has to be stalled because of it as the program will go do other things, most likely further I/O somewhere else.
 
-Async is **not multithreading**. You are still bound to a single execution thread. Async works best when you have a lot of I/O codepaths that don't have to sit there waiting for other I/O requests to end to begin doing their requests.
+Async is **not multithreading** - HHVM still executes all of your PHP/Hack code in one main request thread - but other operations (eg MySQL queries) can now execute without taking up time in that thread that your code could be using.
 
-The following images should hopefully clear up any confusion you may have on what async is. Let's assume you have three distinct tasks to execute (don't worry about what the tasks are):
+## A Page As A Dependency Tree
 
-## Synchronous Execution
+Imagine you have a page that contains two components; one stores data in MySQL, the other fetches from an API via CuRL - and both cache results in Memcached. The dependencies could be modeled like this:
 
-This is just like PHP and Hack (without async) is executed today. Serial execution.
+![Dependency Tree](/images/async/async-dependency.png)
 
-![Synchronous](/images/synchronous.png)
+Code structured like this gets the most benefit from async.
 
-## Parallel Execution
+## Synchronous/Blocking IO: Sequential Execution
 
-This is an optimum state. We have all tasks running at the same time, concurrently. But, PHP and Hack do not support more than one thread of execution.
+If (like most PHP code) you do not use asynchronous programming, each step will be executed one-after-the-other:
```

Comment:
- I'm thinking I'll add an example (with diagram) like running fibonacci + blocking curl + blocking mysql with `HH\Asio\v` to show that it isn't
 - curl and mysqli support asynchronous execution via polling, and things can be cludged on top of this via generators. I picked this phrasing to avoid a flamewar over this, especially as this concept is getting more popular